frame.py

```diff
@@ -0,0 +1,249 @@
+		tmp = Draft.clone(corner)
+		group.addObject(tmp)
+		tmp.Placement = FreeCAD.Placement(FreeCAD.Vector(self.LX,0,0), FreeCAD.Rotation(FreeCAD.Vector(0,1,0),270), FreeCAD.Vector(0,0,0))
+
+		tmp = Draft.clone(corner)
+		group.addObject(tmp)
+		tmp.Placement = FreeCAD.Placement(FreeCAD.Vector(self.LX,self.LY,0), FreeCAD.Rotation(FreeCAD.Vector(0,0,1),180), FreeCAD.Vector(0,0,0))
+		tmp = Draft.clone(corner)
+		group.addObject(tmp)
+		tmp.Placement = FreeCAD.Placement(FreeCAD.Vector(0,self.LY,0), FreeCAD.Rotation(FreeCAD.Vector(0,0,1),270), FreeCAD.Vector(0,0,0))
+
+		tmp = Draft.clone(corner)
+		group.addObject(tmp)
+		tmp.Placement = FreeCAD.Placement(FreeCAD.Vector(0,self.LY, self.LZ), FreeCAD.Rotation(FreeCAD.Vector(1,0,0),180), FreeCAD.Vector(0,0,0))
+		tmp = Draft.clone(corner)
+		group.addObject(tmp)
+		# First rotation.
+		tmp.Placement = FreeCAD.Placement(FreeCAD.Vector(0,0,0), FreeCAD.Rotation(FreeCAD.Vector(0,0,1),180), FreeCAD.Vector(0,0,0))
+		# Second rotation + shift.
+		tmp.Placement = FreeCAD.Placement(FreeCAD.Vector(self.LX,self.LY, self.LZ), FreeCAD.Rotation(FreeCAD.Vector(1,0,0),90), FreeCAD.Vector(0,0,0)).multiply(tmp.Placement)
+		
+	def create(self, convertToSolid):
+		self.checkDimensions()
+		group = self.document.addObject("App::DocumentObjectGroup", "frame box group")
+		self.createPipes(group, convertToSolid)
+		self.addCorners(group, convertToSolid)
+		return group
+
+
+class BoxFromTable:
+	"""Create a part with dimensions from a CSV table."""
+	def __init__ (self, document, pipe_table, corner_table):
+		self.document = document
+		self.pipe_table = pipe_table
+		self.corner_table = corner_table
+		self.LX = tu("12 in")
+		self.LY = tu("10 in")
+		self.LZ = tu("8 in")
+
+	def getCorner(self, partName):
+		corner = outerCorner.OuterCorner(self.document)
+		row = self.corner_table.findPart(partName)
+		if row is None:
+			print('Corner part "%s" not found'%partName)
+			return
+		corner.G = tu(row["G"])
+		corner.H = tu(row["H"])
+		corner.M = tu(row["M"])
+		corner.POD = tu(row["POD"])
+		corner.PID = tu(row["PID"])
+		return corner
+		
+	def create(self, pipeName, cornerName, convertToSolid = True):
+		frame_box = Box(self.document)
+		frame_box.LX = self.LX
+		frame_box.LY = self.LY
+		frame_box.LZ = self.LZ
+		# Init corner datata
+		frame_box.corner = self.getCorner(cornerName)
+		frame_box.G = frame_box.corner.G
+
+		"setup pipe dimensions"
+		row = self.pipe_table.findPart(pipeName)
+		if row is None:
+			print('Pipe part "%s" not found'%pipeName)
+			return
+		frame_box.PID = tu(row["ID"])
+		frame_box.POD = tu(row["OD"])
+		return frame_box.create(convertToSolid)
+
+# Test macros.
+def TestBox():
+	document = FreeCAD.activeDocument()
+	box = Box(document)
+	box.create(True)
+	document.recompute()
+
+def TestTable():
+	document = FreeCAD.activeDocument()
+	pipe_table = CsvTable(PIPE_DIMENSIONS_USED)
+	corner_table = CsvTable(CORNER_DIMENSIONS_USED)
+	pipe_table.load(PIPE_CSV_TABLE_PATH)
```

Comment:
I don't see where these two CSV table paths are defined. Am I missing something?

Reply:
it is a bug. I will fix it